src/sinks/splunk_hec/logs/config.rs

```diff
@@ -140,22 +161,58 @@ pub struct HecLogsSinkConfig {
     //       `None` type and an empty string. This is necessary because `OptionalTargetPath` deserializes an
     //       empty string to a `None` path internally.
     pub timestamp_key: Option<OptionalTargetPath>,
+    #[configurable(derived)]
+    #[serde(default = "default_timestamp_format")]
+    pub format: TimestampFormat,
+    // This settings is relevant only for the `humio_logs` sink and should be left as `None`
+    // everywhere else.
+    #[serde(skip)]
+    pub timestamp_nanos_key: Option<String>,
+    /// Whether to preserve the timestamp field in the event after extraction.
+    pub preserve_timestamp_key: bool,
+}
 
-    /// Passes the `auto_extract_timestamp` option to Splunk.
-    ///
-    /// This option is only relevant to Splunk v8.x and above, and is only applied when
-    /// `endpoint_target` is set to `event`.
-    ///
-    /// Setting this to `true` causes Splunk to extract the timestamp from the message text
-    /// rather than use the timestamp embedded in the event. The timestamp must be in the format
-    /// `yyyy-mm-dd hh:mm:ss`.
-    #[serde(default)]
-    pub auto_extract_timestamp: Option<bool>,
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+/// The format of the timestamp.
+pub enum TimestampFormat {
+    /// Use the default vector time format.
+    Native,
+    /// Numeric timestamp with configurable precision
+    Numeric (TimePrecision),
+    /// `strftime`-style format string to format or parse timestamps.
+    Strftime(String),
```

Comment:
Sorry for the confusion here, a better name would be Strptime (not Strftime), because we'd be parsing not formatting.